File: WeatherClothingApp/weather-clothing-app/src/services/weatherService.js
import axios from "axios";

const WEATHER_API_URL = "http://localhost:5076/api/Weather/GetWeather";

const getWeatherByCity = async (city) => {
    try {
        console.log('Запрос к API с городом:', city);
        const response = await axios.get(`${WEATHER_API_URL}/${city}`);
        console.log(response.data);
        
        if (response.data && response.data.temp && response.data.description && response.data.recommendation) {
            return response.data;
        } else {
            throw new Error("Не удалось получить данные погоды для этого города.");
        }
    } catch (error) {
        console.error("Ошибка определения прогноза погоды для этого города: ", error);
        throw error;
    }
};

export { getWeatherByCity };

File: WeatherClothingApp/Services/WeatherService.cs
namespace WeatherClothingApp.Services
{
    public class WeatherService(HttpClient httpClient)
    {
        private readonly HttpClient _httpClient = httpClient;
        private readonly string _openWeatherApiKey = "391b763977dde18c7f81e3cb8b0b05b2";
        private readonly string _openWeatherBaseUrl = "http://api.openweathermap.org/data/2.5/weather";


        public async Task<WeatherResponse> GetWeatherByCityAsync(string city)
        {
            string url = $"{_openWeatherBaseUrl}?q={city}&appid={_openWeatherApiKey}&units=metric";

            string response = await _httpClient.GetStringAsync(url);

            WeatherResponse weatherData = JsonConvert.DeserializeObject<WeatherResponse>(response) ??  
                throw new HttpRequestException("Error: Data is empty.");

            return weatherData;
        }
    }
}